### PR TITLE
[Addon]feat: add advanced options for fluxcd controllers

### PR DESCRIPTION
**Pull request by ThuWangzw**
### Description of your changes

Add advanced parameters to support [options](https://fluxcd.io/flux/components/helm/options/) for fluxcd's controller.

Example:
``` vela addon enable fluxcd clusters="{local}" helmControllerOptions="{--no-cross-namespace-refs,--concurrent=5}"```

### How has this code been tested?
I have tested on my cluster. 

### Checklist

I have:

- [x] Title of the PR starts with type (e.g. `[Addon]` , `[example]` or `[Doc]`).
- [ ] Updated/Added any relevant [documentation](https://kubevela.io/docs/reference/addons/overview) and [examples](https://github.com/kubevela/catalog/tree/master/examples).
- [ ] New addon should be put in [experimental](https://github.com/kubevela/catalog/tree/master/experimental/addons).
- [x] Update addon should modify the `version` in `metadata.yaml` to generate a new version.

####  Verified Addon promotion rules

If this pr wants to promote an experimental addon to verified, you must check whether meet these conditions too:
  - [ ] This addon must be tested by addon's [e2e-test](./test/e2e-test/addon-test) to guarantee this addon can be enabled successfully.
  - This addon must have some basic but necessary information.
    - [ ] An accessible icon url and source url defined in addon's `metadata.yaml`.
    - [ ] A detail introduction include a basic example about how to use and what's the benefit of this addon in `README.md`.
    - [ ] Also provide an introduction in KubeVela [documentation](https://kubevela.net/docs/reference/addons/overview).
    - [ ] It's more likely to be accepted if useful examples provided in example [dir](examples/).

### CODE REVIEW

File: addons/fluxcd/resources/components/source-controller.cue [hunk 1]
```diff
@@ -6,6 +6,11 @@ controllerArgs: [...]
 _targetNamespace: string
 _sourceControllerName: "fluxcd-source-controller"
 
+imageControllerDefaultArgs: controllerArgs + [
+					"--storage-path=/data",
+					"--storage-adv-addr=http://" + _sourceControllerName + "." + _targetNamespace + ".svc:9090",
+]
+
```
Comment (wangyikewxgm) [inline addons/fluxcd/resources/components/source-controller.cue:13]: Why we need add these config?

File: addons/fluxcd/metadata.yaml [hunk 2]
```diff
@@ -1,5 +1,5 @@
 name: fluxcd
-version: 2.3.1
+version: 2.4.0
```
Comment (wangyikewxgm) [inline addons/fluxcd/metadata.yaml:2]: Bump up to a minor version is enough.

Comment (ThuWangzw) [reply addons/fluxcd/metadata.yaml]: done

File: addons/fluxcd/resources/components/source-controller.cue (hunk 1 above)
Comment (ThuWangzw) [reply addons/fluxcd/resources/components/source-controller.cue:13]: well, these config were not added by me: [link](https://github.com/kubevela/catalog/blame/858842fb9974a25e515d66ffde4fef49bdaad871/addons/fluxcd/resources/components/source-controller.cue#L81)

**Review by wangyikewxgm — APPROVED** [review]
LGTM

**Review by wonderflow — APPROVED** [review]

**Pull request merged**